From 28886f03ffd2e863183cf5de9e43336dc5a3cd9f Mon Sep 17 00:00:00 2001
From: Eero Tamminen <oak@helsinkinet.fi>
Date: Thu, 13 Jun 2024 21:34:08 +0300
Subject: [PATCH 4/4] WA: avoid unexpected SCU interrupts

Avoid freeze on TT / MegaST* boot with continues messages of:
unexpected interrupt from 112
---
 arch/m68k/atari/ataints.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/m68k/atari/ataints.c b/arch/m68k/atari/ataints.c
index 230203ffb..e26af0962 100644
--- a/arch/m68k/atari/ataints.c
+++ b/arch/m68k/atari/ataints.c
@@ -309,6 +309,7 @@ void __init atari_init_IRQ(void)
 									 * enabled in VME mask
 									 */
 		tt_scu.vme_mask = 0x60;		/* enable MFP and SCC ints */
+		vectors[VEC_INT4] = falcon_hblhandler;
 	} else {
 		/* If no SCU and no Hades, the HSYNC interrupt needs to be
 		 * disabled this way. (Else _inthandler in kernel/sys_call.S
-- 
2.39.2

